security/s2a/internal/crypter/common.go

```diff
@@ -0,0 +1,15 @@
+package crypter
+
+// S2AAeadCrypter is the interface for an AEAD cipher used by the S2A record protocol.
+type S2AAeadCrypter interface {
+	// Encrypt encrypts the plaintext and computes the tag of dst
+	// and plaintext. dst and plaintext do not overlap.
```

Comment:
It might be worth clarifying the comment more. I'm assuming you will be using the AEAD object [1] to implement this interface. If that's the case, then the tag is only computed for the plaintext and the result (ciphertext+tag) is appended to the dst. Also, I think dst and plaintext can either fully overlap, or not overlap at all.


[1] https://golang.org/pkg/crypto/cipher/#AEAD

Reply:
I just saw Matt's comment about the comment. We don't have to go in details of reusing the plaintext buffer, but we should say that dst and plaintext can either fully overlap or not at all.

Reply:
Do you think this is worth changing in [ALTSRecordCrypto](https://github.com/grpc/grpc-go/blob/695da821e98cb7a10c41e3f5008aa994560e551a/credentials/alts/internal/conn/record.go#L33) as well? Much of the documentation for the interface was taken from the ALTS interface.

Reply:
That's a good idea. If you would like to send a PR for gRPC-Go to fix the documentation, please ping me directly and I can review it.